frontends/common/options.cpp

```diff
@@ -121,6 +121,9 @@ CompilerOptions::CompilerOptions() : Util::Options(defaultMessage) {
                        }
                        return true; },
                    "Choose output format for the P4Runtime API description (default is binary).");
+    registerOption("--emit-externs", nullptr,
```

Comment:
This option does not belong to common - it should be specific to the back-end.
Also, the documentation has to be more expansive, describing what will actually be generated in the output code.